js/bitget3.js

```diff
@@ -0,0 +1,2899 @@
+const { ExchangeError, ExchangeNotAvailable, OnMaintenance, ArgumentsRequired, BadRequest, AccountSuspended, InvalidAddress, PermissionDenied, DDoSProtection, InsufficientFunds, InvalidNonce, CancelPending, InvalidOrder, OrderNotFound, AuthenticationError, RequestTimeout, NotSupported, BadSymbol, RateLimitExceeded } = require ('./base/errors');
+const { TICK_SIZE } = require ('./base/functions/number');
+const Precise = require ('./base/Precise');
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class bitget3 extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'bitget3',
+            'name': 'Bitget',
+            'countries': [ 'SG' ],
+            'version': 'v3',
+            'rateLimit': 1000, // up to 3000 requests per 5 minutes ≈ 600 requests per minute ≈ 10 requests per second ≈ 100 ms
+            'has': {
+                'cancelOrder': true,
+                'cancelOrders': true,
+                'CORS': undefined,
+                'createOrder': true,
+                'fetchAccounts': true,
+                'fetchBalance': true,
+                'fetchClosedOrders': true,
+                'fetchCurrencies': true,
+                'fetchDeposits': true,
+                'fetchMarkets': true,
+                'fetchMyTrades': true,
+                'fetchOHLCV': true,
+                'fetchOpenOrders': true,
+                'fetchOrder': true,
+                'fetchOrderBook': true,
+                'fetchOrderTrades': true,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTime': true,
+                'fetchTrades': true,
+                'fetchWithdrawals': true,
+            },
+            'timeframes': {
+                '1m': '1m',
+                '5m': '5m',
+                '15m': '15m',
+                '30m': '30m',
+                '1h': '1h',
+                '2h': '2h',
+                '4h': '4h',
+                '6h': '6h',
+                '12h': '12h',
+                '1d': '1d',
+                '1w': '1w',
+            },
+            'hostname': 'bitget.com',
+            'urls': {
+                'logo': 'https://user-images.githubusercontent.com/51840849/88317935-a8a21c80-cd22-11ea-8e2b-4b9fac5975eb.jpg',
+                'api': {
+                    'data': 'https://capi.{hostname}',
+                    'api': 'https://capi.{hostname}',
+                    'capi': 'https://capi.{hostname}',
+                    'swap': 'https://capi.{hostname}',
+                },
+                'www': 'https://www.bitget.com',
+                'doc': [
+                    'https://bitgetlimited.github.io/apidoc/en/swap',
+                    'https://bitgetlimited.github.io/apidoc/en/spot',
+                ],
+                'fees': 'https://www.bitget.com/en/rate',
+                'test': {
+                    'rest': 'https://testnet.bitget.com',
```

Comment:
This testnet seems to be dead (did the link change? I could not find out). Also, the "test" structure must match the "api" structure, otherwise, will crash on the signing when doing `this.implodeHostname`

Reply:
Fixed in PR #11907